In [50]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from matplotlib import pyplot
from statsmodels.stats.proportion import proportions_ztest, proportion_confint, confint_proportions_2indep
from scipy.stats import t, probplot, ttest_1samp, ttest_ind, ttest_rel, sem
from statsmodels.api import qqplot, stats
from statsmodels.formula.api import ols
from math import sqrt
from statsmodels.stats.weightstats import _tstat_generic

n_f = 4208
n_m = 2763
yes_f = 2777
yes_m = 1363
prop_f = yes_f / n_f
prop_m = yes_m / n_m

se_f = prop_f * (1 - prop_f)/ n_f
se_m = prop_m * (1 - prop_m)/ n_m
se_combined = sqrt(se_f + se_m)
print(se_combined)

0.01199154662477283


In [37]:
v , a = confint_proportions_2indep(count1 = yes_f, nobs1 = n_f, count2 = yes_m, nobs2 = n_m,  compare='diff',
                                alpha=0.05, correction = True)
print(v, a)

0.14304339032179628 0.19002232634835703


## Two-Proportion z-interval

In [39]:
# Import the data
titanic = pd.read_csv("titanic_2020.txt", sep = "\t")

In [40]:
# Look at the first few rows
titanic.head()

Name    Born    Died   Age  Gender  \
0                            ABBING, Mr Anthony  1870.0  1912.0  41.0    Male   
1                      ABBOTT, Mr Eugene Joseph  1899.0  1912.0  13.0    Male   
2                    ABBOTT, Mr Rossmore Edward  1896.0  1912.0  16.0    Male   
3                 ABBOTT, Mrs Rhoda Mary 'Rosa'  1873.0  1946.0  39.0  Female   
4  ABELSETH, Miss Kalle (Karen) Marie Kristiane  1895.0  1969.0  16.0  Female   

            Class.Dept Class  Ticket                Fare Cabin  ...  \
0  3rd Class Passenger   3rd    5547  £7             11s   NaN  ...   
1  3rd Class Passenger   3rd  CA2673  £20             5s   NaN  ...   
2  3rd Class Passenger   3rd  CA2673  £20             5s   NaN  ...   
3  3rd Class Passenger   3rd  CA2673  £20             5s   NaN  ...   
4  3rd Class Passenger   3rd  348125  £7             13s   NaN  ...   

   Occupation Survived Boat   Body  Nationality Pounds  Shillings  Pence  \
0  Blacksmith     LOST  NaN    NaN     American    7.0       11.0    NaN   
1     Scholar     LOST  NaN    NaN     American   20.0        5.0    NaN   
2    Jeweller     LOST  NaN  190.0          NaN   20.0        5.0    NaN   
3         NaN    SAVED    A    NaN      English   20.0        5.0    NaN   
4         NaN    SAVED   16    NaN          NaN    7.0       13.0    NaN   

   Price.Paid  Paidpp  
0        7.55    7.55  
1       20.25    6.75  
2       20.25    6.75  
3       20.25    6.75  
4        7.65    7.65  

[5 rows x 21 columns]

In [44]:
pd.crosstab(titanic["Gender"], titanic["Survived"], margins = True)

Survived  LOST  SAVED   All
Gender                     
Female     130    359   489
Male      1366    353  1719
All       1496    712  2208

In [47]:
lower , upper = confint_proportions_2indep(count1 = 359, nobs1 = 489, count2 = 353, nobs2 = 1719,  compare='diff',
                                alpha=0.05, correction = True)
print(lower, upper)

0.4834092767839217 0.5703331209312256


We are 95% confident that the proportion of female who survided in the Titanic is between 48.34% and 57% higher than the proportion of male who survived.

In [52]:
female = 39667.2
male =  46484.0


In [54]:
_tstat_generic(female, male, -1573.9, 1773, alternative = "two-sided")

(4.331151915623612, 1.566296274045779e-05)